src/test/java/nc/students/ayaz/controllers/VideoControllerTest.java

```diff
@@ -0,0 +1,123 @@
+package nc.students.ayaz.controllers;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import nc.students.ayaz.deserializer.VideoDeserializer;
+import nc.students.ayaz.model.Comment;
+import nc.students.ayaz.model.User;
+import nc.students.ayaz.model.Video;
+import nc.students.ayaz.model.exceptions.NoSuchUserException;
+import nc.students.ayaz.repositories.UserRepository;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class VideoControllerTest {
+
+    @MockBean
+    private UserRepository repository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    private ObjectMapper mapper;
+
+    @Before
+    public void setup() {
+        this.mapper = new ObjectMapper();
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(Video.class, new VideoDeserializer());
+        mapper.registerModule(module);
+    }
+
+    @Test
+    public void shouldAddVideoToUserWhenPostRequestSentWithVideoname() throws Exception {
```

Comment:
&#x1F34E; **Issue**
There is no such thing as Videoname ;)
shouldAddVideoToUserWhenPostRequestSentWithVideoname -> shouldAddVideoToUserWhenPostRequestSentWithVideoName